src/treeherder/action_handler.js

```diff
@@ -42,8 +42,9 @@ class Handler {
   }
 
   async handleRetrigger(taskId, runId, task, payload) {
-    // Must be a scheduled task and have a task group ID
-    if (task.schedulerId !== SCHEDULER_TYPE || !task.taskGroupId) {
+    // Must have a task group ID
+    if (!task.taskGroupId) {
```

Comment:
If loaded from the queue it **always** has a `taskGroupId`. It's a required property, see docs...

(not required for creation, but always given when returned, it defaults to the taskId if a taskGroupId isn't given).